### ISSUE TITLE
Computed properties still empty, when state is changed

### ISSUE THREAD

**Issue by meliborn** [maoberlehner/vuex-map-fields#41]
Not sure it's a vuex-map-fields problem but...
I have a simple form and submit button. I do axios call to send data to the server. The problem is, when I type something into form input, property still empty, when store is up to date. 
I'm using chrome plugin to debug values:
![111](https://user-images.githubusercontent.com/735194/45309199-df565b80-b52b-11e8-9ab3-f2294764a40d.png)
Here is a store:

![222](https://user-images.githubusercontent.com/735194/45309230-f137fe80-b52b-11e8-9d86-3d5f1bf13286.png)
So when I submit the form it always clear filled values. What's wrong?

**Comment by maoberlehner** [comment]
Hey @meliborn,

without seeing any code, there is nothing I can do for you ;)

Please provide a [reduced test case](https://css-tricks.com/reduced-test-cases/) (you can use https://codesandbox.io/ for example) or at least copy and paste the relevant code (the `computed` section of your component and the `getters` and `mutations` sections of your store). Thanks!

**Comment by meliborn** [comment]
@maoberlehner 
```
computed: {
    companyExist(){
      return this.$store.state.company.current.id;
    },
    ...mapFields('company', [
      'current.name',
      'current.description',
      'current.company_type',
    ]),
  },
```

```
// initial state
import { getField, updateField } from "vuex-map-fields";
import { LOAD_COMPANY } from "@/store/action-types";
import { SET_COMPANY } from "@/store/mutation-types";
import axios from "axios";

export default {
  namespaced: true,
  state: {
    current: {}
  },
  getters: {
    getField
  },
  mutations: {
    updateField,
    [SET_COMPANY](state, company){
      state.current = company;
    }
  },
  actions: {
    [LOAD_COMPANY](state) {
      axios.get("/company").then(company => {
        state.commit(SET_COMPANY, company.data);
      });
    }
  }
};
```

**Comment by anikolaev** [comment]
It could look like this:
```js
function updateField(state, { path, value }, makeReactive = false) {
  path.split(/[.[\]]+/).reduce((prev, key, index, array) => {
    if (array.length === index + 1) {
      if (makeReactive && !prev.hasOwnProperty(key)) {
        Vue.set(prev, key, value)
      } else {
        prev[key] = value
      }
    }

    return prev[key]
  }, state)
}
```
I used it instead of librarian one and it works for plain and object properties. Haven't checked with arrays yet.

**Comment by maoberlehner** [comment]
This could be an interesting addition to the code. I'd take a look at a pull request if somebody invests their time into that. 👍

**Comment by Uelb** [comment]
Is it possible to add this case to the readme for now ? I actually struggled to understand where the issue was and I faced the exact same case. If I have time next week, I'll try to open a PR with a solution.

**Comment by smith153** [comment]
Yes, this would be a great feature. I have many forms that are 100+ variables, really don't want to have to add defaults for every single one (and whether some fields are even shown depends on values of other variables, so you really don't have a concept of "defaults" in this case). Seems like an easy fix... though I suppose we could just wait for vue 2.6+ as I hear the requirement of forcing defaults will be removed...

**Comment by leewisestamp** [comment]
Eagerly looking forward to this feature!

**Comment by jdinartejesus** [comment]
Same here, I'm currently having the same issue even after update to V1.4.0.
